src/utils/inputValidation/__tests__/attributes.spec.ts

```diff
@@ -49,3 +49,40 @@ describe('INPUT VALIDATION for Attributes', () => {
     expect(loggerMock.warn).not.toBeCalled(); // It should have not logged any warnings.
   });
 });
+
+describe('DEEP INPUT VALIDATION for Attributes', () => {
+
+  afterEach(() => { loggerMock.mockClear(); });
+
+  test('Should return true if it is a valid attributes map without logging any errors', () => {
+    const validAttributes = { amIvalid: 'yes', 'are_you_sure': true, howMuch: 10, 'spell': ['1', '0'] };
+
+    expect(validateAttributesDeep(loggerMock, validAttributes, 'some_method_attrs')).toEqual(true); // It should return true if it is valid.
+    expect(loggerMock.error).not.toBeCalled(); // Should not log any errors.
+    expect(loggerMock.warn).not.toBeCalled(); // It should have not logged any warnings.
+
+  });
+
+  test('Should return false and log error if attributes map is invalid', () => {
+
+    expect(validateAttributesDeep(loggerMock, { '': 'empty' }, 'some_method_attrs')).toEqual(false); // It should be invalid if the attribute key is not a string
+    expect(validateAttributesDeep(loggerMock, { 'attributeKey': new Date() }, 'some_method_attrs')).toEqual(false); // It should be invalid if the attribute value is not a String, Number, Boolean or Lists.
```

Comment:
migrate the tape assert `assert.equals(validateAttributesDeep({'attrKey': null}, 'some_method_attrs'), false, 'should be invalid if the attribute value is not a String, Number, Boolean or Lists.');`